src/App.tsx

```diff
@@ -1,39 +1,56 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  today: Date;
+  clockName: string;
+  hasClock: boolean;
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  state = {
+    today: new Date(),
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  render() {
+    const { clockName, today, hasClock } = this.state;
+
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+
+      this.setState({ hasClock: false });
+    });
+
+    document.addEventListener('click', () => {
+      this.setState({
+        hasClock: true,
+        clockName: getRandomName(),
+        today: new Date(),
+      });
+    });
```

Comment:
you should add event listeners in componentDidMount and remove them in component will unmount